### PR TITLE
Add documentation for website security

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Fixed tickets | ---
| Related PRs | sulu/sulu#5560
| License | MIT

#### What's in this PR?

This PR adds documentation for the new website security.

### CODE REVIEW

File: cookbook/user-context-caching.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            enabled: true
+
+5. Make sure that Symfony's ``SecurityBundle`` is registered for all contexts
+   (by default it is only registered for the admin context) in `config/bundles.php`
+
+.. code-block:: php
+
+    Symfony\Bundle\SecurityBundle\SecurityBundle::class => ['all' => true],
+
+6. Add the login and logout routes to `config/routes/sulu_website.yaml`
```
Comment (alexander-schranz) [inline cookbook/user-context-caching.rst:101]: The `config/routes/*` folder contains routes of bundles. For app specific routes the `config/routes.yaml` should be used or in our case `config/routes_website.yaml`:

```suggestion
6. Add the login and logout routes to `config/routes_website.yaml`
```

File: cookbook/user-context-caching.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            Sulu\Bundle\SecurityBundle\Entity\User: bcrypt
+
+        providers:
+            sulu:
+                id: sulu_security.user_provider
+
+        firewalls:
+            website:
+                pattern: ^/
+                anonymous: lazy
```
Comment (alexander-schranz) [inline cookbook/user-context-caching.rst:78]: This option was splitted maybe need to be updated in the skeleton:

```suggestion
                anonymous: true
                lazy: true
```

File: bundles/media/security.rst [hunk 3]
```diff
@@ -4,11 +4,17 @@ Security
 Separate permissions can be applied to every collection. These permissions also
 apply to the assets contained in the collection.
 
-So the `view` permission is required to see the collection itself in the
-navigation and its assets in any available view in the system.
+So in the Sulu system (representing Sulu's administration interface) the `view`
```
Comment (alexander-schranz) [inline bundles/media/security.rst:7]: Sadly I think this is not longer true as now I need `edit` permission to see the Navigation Item:

https://github.com/sulu/sulu/blob/ae5b58ea48ebd260eb762295ef483add46d697ae/src/Sulu/Bundle/MediaBundle/Admin/MediaAdmin.php#L83

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Some small change requests.

File: book/webspaces.rst [hunk 4]
```diff
@@ -113,6 +117,28 @@ internally to generate some files and define some paths. Therefore it is really
...
+If the ``permission-check`` attribute of the ``security`` tag is set to
+``true``, Sulu will automatically check if the current user has access to see
+the requested page. It will also make sure that no pages are listed in the
+website's navigation or in smart contents, if the user does not have the
+necessary permissions.
+
+.. note::
+
+    Make sure caching is set up correctly if you use the security feature with
+    the ``permission-check`` flag set to false. The caching will only work if
```
Comment (nnatter) [inline book/webspaces.rst:139]: shouldnt the be `flag set to true`?

File: bundles/media/security.rst (hunk 3 above)
Comment (danrot) [reply bundles/media/security.rst:7]: I am talking about the permissions on the collections themselves... There you don't need the edit permission to see the collections in the UI.

File: book/webspaces.rst (hunk 4 above)
Comment (danrot) [reply book/webspaces.rst:139]: Right

**Pull request merged**